MathLib/LinAlg/PETSc/PETScVector.cpp

```diff
@@ -101,6 +119,13 @@ void PETScVector::finalizeAssembly()
 {
     VecAssemblyBegin(_v);
     VecAssemblyEnd(_v);
+
+    if (!_global_v)
```

Comment:
Is this necessary here?
AFAIK finalized matrices/vectors are only used by the solvers which don't need to access ghost entries.